### PR TITLE
Fix windows debug build

### PR DESCRIPTION
**Pull request by cyyever**
links to the python debug library

### CODE REVIEW

File: caffe2/CMakeLists.txt [hunk 1]
```diff
@@ -1314,7 +1314,12 @@ if (BUILD_PYTHON)
   target_link_libraries(
       caffe2_pybind11_state torch_library)
   if (WIN32)
-    target_link_libraries(caffe2_pybind11_state ${PYTHON_LIBRARIES})
+    if ((GENERATOR_IS_MULTI_CONFIG AND "Debug" IN_LIST CMAKE_CONFIGURATION_TYPES)
+      OR ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+      target_link_libraries(caffe2_pybind11_state ${PYTHON_DEBUG_LIBRARIES})
```
Comment (peterjc123) [inline caffe2/CMakeLists.txt]: I guess a better solution for multiple-build-type generators is to use generator expressions.
Should be sth like `target_link_libraries(caffe2_pybind11_state  $<$<CONFIG:Debug>:${PYTHON_DEBUG_LIBRARIES}> $<$<NOT:$<CONFIG:Debug>>:${PYTHON_LIBRARIES}>)`.

Comment (cyyever) [reply caffe2/CMakeLists.txt]: We need to change these conditions thoroughly.

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -839,6 +839,12 @@ else()
             DESTINATION ${CMAKE_INSTALL_PREFIX}
             FILES_MATCHING PATTERN "*.h")
 endif()
+if (MSVC)
+  if ((GENERATOR_IS_MULTI_CONFIG AND "Debug" IN_LIST CMAKE_CONFIGURATION_TYPES)
+      OR ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+    add_compile_definitions(Py_DEBUG)
```
Comment (peterjc123) [inline cmake/Dependencies.cmake]: This is not needed. When `_DEBUG` is defined, `Py_DEBUG` is defined too. The logic is in `pyconfig.h`.

Comment (cyyever) [reply cmake/Dependencies.cmake]: > This is not needed. When _DEBUG is defined, Py_DEBUG is defined too. The logic is in pyconfig.h.

pybind11 undef _DEBUG deliberately

Comment (cyyever) [reply cmake/Dependencies.cmake]: > This is not needed. When _DEBUG is defined, Py_DEBUG is defined too. The logic is in pyconfig.h.
> 
> pybind11 undef _DEBUG deliberately

see this https://github.com/pybind/pybind11/issues/1295

Comment (peterjc123) [reply cmake/Dependencies.cmake]: Okay.

Comment (cyyever) [reply cmake/Dependencies.cmake]: > Okay.

I use this macro for system pybind11 and for the updated 3rd module in future .

**Pull request closed without merging**